src/measure_ops.jl

```diff
@@ -19,6 +19,50 @@ function eigenbasis(op::AbstractBlock{N}) where {N}
     end
 end
 
+for BT in []
+    @eval function eigenbasis(op::$BT)
+        @warn "eigenbasis on blocktype `$($BT)` calls into the fallback implementation, which might be slow. Try using `kron`, `repeat` if items commute to each oher."
+        invoke(eigenbasis, Tuple{AbstractBlock}, op)
+    end
+end
+
+"""
+Return true if operators commute to each other.
+"""
+function simple_commute_eachother(ops::Vector{<:AbstractBlock{N}}) where N
+    occ = zeros(Bool, N)
+    for op in ops
+        for i in occupied_locs(op)
+            if occ[i]
+                return false
+            else
+                occ[i] = true
+            end
+        end
+    end
+    return true
+end
+
+function eigenbasis(op::ChainBlock{N}) where N
```

Comment:
@wangleiphy Is this ok?

Reply:
It is OK! I tested it in my application. Works and is faster. 
👍